### PR TITLE
Add kubernetes deployment yaml files

### PR DESCRIPTION
**Pull request by Mmduh-483**

### CODE REVIEW

File: deployment/ib-kubernetes-pre-1.16.yaml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: ib-kubernetes
+rules:
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get", "list", "watch", "patch", "update"]
```
Comment (moshe010) [inline deployment/ib-kubernetes-pre-1.16.yaml]: do we need "update" I thought we are just doing patch?
Do we need "list" and "watch"?

File: deployment/ib-kubernetes-pre-1.16.yaml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "list"]
+  - apiGroups: ["k8s.cni.cncf.io"]
```
Comment (moshe010) [inline deployment/ib-kubernetes-pre-1.16.yaml]: is this for CRD?

File: deployment/ib-kubernetes-pre-1.16.yaml [hunk 3]
```diff
@@ -0,0 +1,98 @@
+---
+apiVersion: v1
```
Comment (moshe010) [inline deployment/ib-kubernetes-pre-1.16.yaml]: what the diff between pre-1.16 to the other?

File: deployment/ib-kubernetes-pre-1.16.yaml (hunk 2 above)
Comment (Mmduh-483) [reply deployment/ib-kubernetes-pre-1.16.yaml]: yes

File: deployment/ib-kubernetes-pre-1.16.yaml (hunk 1 above)
Comment (Mmduh-483) [reply deployment/ib-kubernetes-pre-1.16.yaml]: watch and update are not needed so removed, but list is needed when start the daemon to list all the pods and check the ones has guids and mark them as reserved.

File: deployment/ib-kubernetes-pre-1.16.yaml (hunk 3 above)
Comment (Mmduh-483) [reply deployment/ib-kubernetes-pre-1.16.yaml]: Role, ClusterRoleBinding, and Deployment objects needs uses the v1beta1 version while the other uses v1. Seems k8s added comparability in older versions to support api v1, but not sure if is in all older version (tested on 1.15 and 1.14).

**Pull request closed without merging**